upload.go

```diff
@@ -171,12 +185,22 @@ func (tu *TarUploader) upload(input *s3manager.UploadInput, path string) (err er
 // createUploadInput creates a s3manager.UploadInput for a TarUploader using
 // the specified path and reader.
 func (tu *TarUploader) createUploadInput(path string, reader io.Reader) *s3manager.UploadInput {
-	return &s3manager.UploadInput{
+	uploadInput := &s3manager.UploadInput{
 		Bucket:       aws.String(tu.bucket),
 		Key:          aws.String(path),
 		Body:         reader,
 		StorageClass: aws.String(tu.StorageClass),
 	}
+
+	if tu.ServerSideEncryption != "" {
+		uploadInput.ServerSideEncryption = aws.String(tu.ServerSideEncryption)
+
+		if tu.ServerSideEncryption == "aws:kms" && tu.SSEKMSKeyId != "" {
```

Comment:
SSEKMSKeyId should be setted only for ServerSideEncryption, right?
I'm a bit worried about checking string literal "aws:kms" here, validation logic is duplicated now... but probably it is OK.